pytest_mh/utils/firewall.py

```diff
@@ -115,101 +257,415 @@ def teardown_when_used(self):
+
         :param rule: Firewalld rich rule.
         :type rule: str
+        :param priority: Rule priority, defaults to None (= auto-assign next
+            value)
+        :type priority: int | None, optional
+        :return: Rule priority, to be used for rule removal.
+        :rtype: int
         """
+        if priority is None:
+            priority = self._next_priority
+
+        rule = f"rule priority={priority} {rule}"
         self.logger.info(f'Firewalld: adding rich rule "{rule}"')
         self.host.ssh.exec(["firewall-cmd", "--add-rich-rule", rule], log_level=SSHLog.Error)
 
-    def remove_rich_rule(self, rule: str) -> None:
+        return priority
+
+    def remove_rich_rule(self, priority: int, rule: str) -> None:
         """
         Remove rich rule.
 
+        The parameter "rule" is the part after "rule priority=X". This part is
+        added automatically. That is:
+
+        .. code-block:: console
+
+            $ firewall-cmd --remove-rich-rule rule priority="{priority}" {rule}
+
+        :param priority: Rule priority
+        :type priority: int
         :param rule: Firewalld rich rule.
         :type rule: str
         """
+        rule = f"rule priority={priority} {rule}"
         self.logger.info(f'Firewalld: removing rich rule  "{rule}"')
         self.host.ssh.exec(["firewall-cmd", "--remove-rich-rule", rule], log_level=SSHLog.Error)
 
-    def __add_action(
+
+class FirewalldInboundRules(FirewallInboundRules):
+    """
+    Configure firewall inbound rules using firewalld.
+
+    All changes are automatically reverted when a test is finished.
+    """
+
+    def __init__(self, firewall: Firewalld) -> None:
+        """
+        :param firewall: Firewalld controller.
+        :type firewall: Firewalld
+        """
+        super().__init__()
+
+        self.firewall: Firewalld = firewall
+
+    def accept_port(self, port_or_service: ServiceSpec | list[ServiceSpec]) -> None:  # type: ignore[override]
+        """
+        Accept incoming traffic on given port or service.
+
+        .. code-block:: python
+
+            firewall.inbound.accept_port(389)  # missing protocol defaults to "tcp"
+            firewall.inbound.accept_port((389, "tcp"))
+            firewall.inbound.accept_port("ldap")
+            firewall.inbound.accept_port(["ldap", "ldaps"])
+
+        :param port_or_service: Port (integer), (port, protocol) tuple or service (string).
+        :type port_or_service: ServiceSpec | list[ServiceSpec]
+        """
+        self.__add_port(port_or_service, action="accept")
+
+    def reject_port(self, port_or_service: ServiceSpec | list[ServiceSpec]) -> None:
+        """
+        Reject incoming traffic on given port or service.
+
+        .. code-block:: python
+
+            firewall.inbound.reject_port(389)  # missing protocol defaults to "tcp"
+            firewall.inbound.reject_port((389, "tcp"))
+            firewall.inbound.reject_port("ldap")
```

Comment:
This is a lot of repetition. Maybe, three examples for the first function in the class, then any additional function one example?